packages/wallet/contract/src/smartWallet.js

```diff
@@ -1,90 +1,425 @@
+  observeIteration(E(bank).getAssetSubscription(), {
+    async updateState(desc) {
+      await addIssuer({
+        brand: desc.brand,
+        issuer: desc.issuer,
+        petname: desc.proposedName,
+      });
+    },
+  });
+
+  /**
+   * Similar to {DepositFacet} but async because it has to look up the purse.
+   */
+  const depositFacet = Far('smart wallet deposit facet', {
+    /**
+     * Put the assets from the payment into the appropriate purse
+     *
+     * @param {ERef<Payment>} paymentE
+     * @param {Brand} paymentBrand must match the payment's brand. passed in to save lookup.
+     * @returns {Promise<Amount>}
+     * @throws if the purse doesn't exist
+     * ??? PRODUCT: should it instead wait for an issuer/purse to put it in?
+     */
+    receive: (paymentE, paymentBrand) => {
+      const purse = brandPurses.get(paymentBrand);
+
+      const depositP = E.when(paymentE, payment => E(purse).deposit(payment));
+
+      E.when(depositP, publishPurses, () =>
+        console.error('receive() failed to publish updates'),
+      );
+      return depositP;
+    },
+  });
+
+  // XXX refactor the invitation fetching
+  /**
+   * @param {import('./invitations.js').InvitationsPurseQuery} query
+   * @returns {Promise<Invitation>}
+   */
+  const findInvitationInPurse = async query => {
+    const purse = brandPurses.get(invitationBrand);
+    const purseAmount = await E(purse).getCurrentAmount();
+    const match = AmountMath.getValue(invitationBrand, purseAmount).find(
+      details => details.description === query.description,
+      // FIXME match instance too
+    );
+    console.log('DEBUG findInvitationInPurse', { purseAmount, query, match });
+    assert(match, `no match for query ${query} `);
+    const toWithDraw = AmountMath.make(invitationBrand, harden([match]));
+    console.log('.... ', { toWithDraw });
+
+    return E(purse).withdraw(toWithDraw);
+  };
+  const invitationFromSpec = makeInvitationsHelper(
+    zoe,
+    findInvitationInPurse,
+    offerToInvitationMakers.get,
+  );
+
+  /* Design notes:
+   * - an ocap to suggestIssuer() should not have authority to withdraw from purses (so no executeOffer() or applyMethod())
+   * XXX refactor for POLA
+   */
+  /**
+   * Holders of this object:
+   * - vat (transitively from holding the wallet factory)
+   * - wallet-ui (which has key material; dapps use wallet-ui to propose actions)
+   */
   return Far('SmartWallet', {
-    ...wallet,
-    getSubscriber: () => storedSubscriber,
-    performAction: obj => E(admin).performAction(obj),
+    /**
+     * Take an offer description provided in capData, augment it with payments and call zoe.offer()
+     *
+     * @param {WalletCapData<OfferSpec>} capData
+     * @returns {void} when the offer has been sent to Zoe; payouts go into this wallet's purses
+     * @throws if any parts of the offer can be determined synchronously to be invalid
+     */
+    executeOffer: capData => {
```

Comment:
I guess this purposely doesn't handle attestations for now right?

Reply:
yeah, not in scope for PSM launch. @dckc will supporting that affect the API?

Reply:
@samsiegart remind me how we represented "this offer should include an attestation" in our `acceptOffer` argument?